### ISSUE TITLE
4.1.1 release is corrupt for MacOS

### ISSUE THREAD

**Issue by txjohnson** [Bforartists/Bforartists#4330]
**Describe the bug**
MacOS Sonoma reports that app extracted from Bforartists-4.1.1-mac-silicon.zip is damaged.
This is running on an M1 MacBook Pro. The exact message:

`“Bforartists 4.1.1 mac silicon.app” is damaged and can’t be opened. You should move it to the Trash.`

This happens when I try to launch after extracting. I've managed to build for Arm64 target on that same MacBook. My build launches with no error.

**Comment by ReinerBforartists** [comment]
> Yeah, it's a known issue, will have a look.

Wasn't there an experiment going on with a script?

**Comment by Blenux** [comment]
> > Yeah, it's a known issue, will have a look.
> 
> Wasn't there an experiment going on with a script?

Trying out now :smile:

**Comment by txjohnson** [comment]
> Yeah, it's a known issue, will have a look.

Ok thanks. 

Just a quick note for those building for Arm64 on MacOS until a solution is found. To build the Arm64 target, you must have an Am64 version of Python in your path. It seems x86 Python under Rosetta will report the wrong platform. Next, 'make upgrade' will pull in appropriate libs for Arm64. Then 'make' will build. 'make release' still looks for x86 libs. I haven't looked any deeper into that.

**Comment by Blenux** [comment]
I ran this, 

python3 -c 'import platform; print(platform.platform())'

and got

**macOS-14.4.1-arm64-arm-64bit**

**Comment by Blenux** [comment]
@txjohnson are you around to test?

**Comment by ReinerBforartists** [comment]
Any news here?

**Comment by Draise14** [comment]
Looks like @Blenux has a build to test, but we need a tester.

**Comment by Draise14** [comment]
There is a tester on Discord looking for a solution.

**Comment by ReinerBforartists** [comment]
You should connect these two ...

**Comment by Draise14** [comment]
Connected, test build failed again. Still needs work...

**Comment by ReinerBforartists** [comment]
Any progress here?

**Comment by Draise14** [comment]
I think the universal build may be fixing this. Requires permission but works. 

@Blenux can confirm? The universal build is how we move forward for Mac from now on?

**Comment by Blenux** [comment]
Yeah the universal build should fix this, really need @txjohnson to test on their end to also confirm (shaley on our discord has confirmed the universal build for him is working at least),

and has draise mentioned, still need to do the xattr -cr after install to let it open until we get a Apple Developer Certificate to officially sign them (which I wanna do soon).

**Comment by ReinerBforartists** [comment]
Okay, so let's keep this issue open until the certificate is in place :)

**Comment by ReinerBforartists** [comment]
Any news at the certificate?

**Comment by txjohnson** [comment]
Hi all, its been a while...

I can confirm that the universal build does work. I've tried on Sonoma through 14.7.4 and Sequoia through 15.3.1. Hope this helps others.

**Comment by Draise14** [comment]
I guess we can close now that the universal build helps.

The certificate issue can be another.
